### DISCUSSION TITLE
Providers Config

### DISCUSSION THREAD

**Discussion by okhaimie-dev** [family/connectkit#285] [Q&A] [answered]
How do I set up different network for the same provider?

So, I am trying to build dApp that should allow users to interact with multiple blockchain, but I am confused how I can have multiple network setup as my providers when I am using the `createConfig`.

```tsx
const config = createConfig(
    getDefaultConfig({
        alchemyId: process.env.NEXT_SERVER_ALCHEMY_ID, // or infuraId
        walletConnectProjectId:
            process.env.NEXT_PUBLIC_WALLETCONNECT_PROJECT_ID || "",
        chains,

        appName: "App UI",
    })
);
```

Do I write a conditional statement that let's me select `alchemyId`? I am not even sure how that will work because you need the provider to be connected first before you know what chain you are on...

Does anyone know my options here?

**Comment by lochie** [comment]
You should be able to use both `alchemyId` and `infuraId` in the config if you require both providers and it should automatically prioritise which provider to use based on their settings. You can learn more about [multiple providers in the wagmi docs](https://wagmi.sh/react/providers/configuring-chains#multiple-rpc-providers).

You can read more about providers in the ConnectKit Docs here: https://docs.family.co/connectkit/providers

**Comment by okhaimie-dev** [comment]
Not multiple providers.

I am talk about multiple network.
<img width="1240" alt="CleanShot 2023-08-15 at 01 05 48@2x" src="https://github.com/family/connectkit/assets/57156589/458cc65e-d6e5-471b-8377-3cd69e50d0d9">

E.g.: Alchemy has different API keys for different networks, I was curious how to switch based on network, but I think I have the answer now, I just have to write a conditional statement.

Thanks for your time

**Reply by lochie** [reply, marked as answer]
Just reading through some of the [wagmi discussions](https://github.com/wagmi-dev/wagmi/discussions) it looks like this has been answered here: https://github.com/wagmi-dev/wagmi/discussions/2805#discussioncomment-6546932
